pytest_fixtures/satellite_auth.py

```diff
@@ -248,48 +250,70 @@ def enroll_configure_rhsso_external_auth():
     audience_mapper['config']['included.client.audience'] = audience_mapper['config'][
         'included.client.audience'
-    ].format(rhsso_host=settings.server.hostname)
+    ].format(rhsso_host=default_sat)
     create_mapper(audience_mapper, client_id)
     set_the_redirect_uri()
 
 
 @pytest.mark.external_auth
 @pytest.fixture(scope='session')
-def enroll_idm_and_configure_external_auth():
+def enroll_idm_and_configure_external_auth(default_sat):
     """Enroll the Satellite6 Server to an IDM Server."""
     run_command(cmd='yum -y --disableplugin=foreman-protector install ipa-client ipa-admintools')
-
     run_command(
         cmd=f'echo {settings.ipa.password} | kinit admin',
         hostname=settings.ipa.hostname,
     )
-    result = run_command(
-        cmd=f'ipa host-add --random {settings.server.hostname}',
-        hostname=settings.ipa.hostname,
-    )
+    output = ssh.command(f'ipa host-find {default_sat.hostname}')
+    if output.return_code != 0:
+        result = run_command(
+            cmd=f'ipa host-add --random {default_sat}',
+            hostname=settings.ipa.hostname,
+        )
+        for line in result:
+            if 'Random password' in line:
+                _, password = line.split(': ', 2)
+                break
+        ssh.command(
+            cmd=f'ipa service-add HTTP/{default_sat}',
+            hostname=settings.ipa.hostname,
+        )
+        _, domain = settings.ipa.hostname.split('.', 1)
+        result = ssh.command(
+            cmd=f"ipa-client-install --password '{password}' "
+            f'--domain {domain} '
+            f'--server {settings.ipa.hostname} '
+            f'--realm {domain.upper()} -U'
+        )
+        if result.return_code not in [0, 3]:
+            CLIReturnCodeError(
+                result.return_code,
+                result.stderr,
+            )
 
-    for line in result:
-        if 'Random password' in line:
-            _, password = line.split(': ', 2)
-            break
-    run_command(
-        cmd=f'ipa service-add HTTP/{settings.server.hostname}',
-        hostname=settings.ipa.hostname,
-    )
-    _, domain = settings.ipa.hostname.split('.', 1)
+
+@pytest.fixture(scope='session')
+def configure_realm():
+    """Configure realm"""
+    realm = settings.upgrade.vm_domain.upper()
+    run_command(cmd=f'wget -O /root/freeipa.keytab {settings.ipa.keytab_url}')
+    run_command(cmd='mv /root/freeipa.keytab /etc/foreman-proxy')
+    run_command(cmd='chown foreman-proxy:foreman-proxy /etc/foreman-proxy/freeipa.keytab')
     run_command(
-        cmd=f"ipa-client-install --password '{password}' "
-        f'--domain {domain} '
-        f'--server {settings.ipa.hostname} '
-        f'--realm {domain.upper()} -U'
+        cmd='satellite-installer --foreman-proxy-realm true '
+        f'--foreman-proxy-realm-principal realm-proxy@{realm} '
+        f'--foreman-proxy-dhcp-nameservers {settings.ipa.idm_server_ip}'
```

Comment:
Please change to this:
`f'--foreman-proxy-dhcp-nameservers {socket.gethostbyname(settings.ipa.hostname)}'`